### ISSUE TITLE
User data verification page is auto-upgrading attendees to the next higher level when their level is sold out or closed

### ISSUE THREAD

**Issue by adebus** [MidwestFurryFandom/rams#374]
For the first time ever, MFF sold out of Sponsor Registrations prior to the convention opening. This has resulted in a number of situations where the attendee accesses their profile to update an address, phone number, or badge name, and gets automatically upgraded to the next higher level of membership with available spots. 

For example, a Sponsor goes in and updates their address, and is then presented with a screen saying they now owe $150. There are numerous tickets which have forced the Registration team to go in and manually fix these issues. 

However, now that the user has "upgraded" that opens a sponsor spot, and frequently another attendee snapped up the "available" sponsorship. This has resulted in MFF being oversold on Sponsorships, needing to cannibalize the Shiny Sponsor gift selections for Sponsor gifts, and has cost the convention $150 for each oversold Sponsorship. 

This is a critical issue that must be fixed prior to pre-reg opening for the 2020 convention.

**Comment by kitsuta** [comment]
Whoa, that's really serious. More than happy to comp this fix given the unexpectedly bad repercussions.

In the future, we should try to figure out the best way to keep communication lines open between dev and registration for this kind of issue. I'd much rather catch and fix this ASAP than have a thoroughly-written ticket (though in this case I would _not_ have guessed at the ability for people to buy the 'open' Sponsor spots).

**Comment by adebus** [comment]
We only discovered it about 2 days before Pre-registration closed,  and we've been going at con prep non-stop, this the first chance I've had to sit down and document the issue. 

I definitely agree that the lines of communication need to be improved. 

We've worked around the issue by temporarily reducing the number of Shiny Sponsorships to the precise number we've sold, and we'll re-increase them to what we have left to sell once we open at-the-door registration.

The workflow here is this:

1) User1 loads their profile page, and makes an update.
2) Because there are no Sponsorships left to buy, the page pre-selects "Shiny Sponsor" because it's the only option available.
3) User1 submits the update, and RAMS removes their Sponsor Designation, replacing it with Shiny Sponsor. (Sponsor-- / Shiny Sponsor++) 
4) User1 is presented with a balance due of $150 (if they even notice, some didn't), and puts in a ticet going "wait, what?"
5) Before one of the 4 full time reg staff can fix it, User2 starts a new registration, and decides to become a sponsor. 
6) User2 finishes their registration (Sponsor++)
7) Reg staff attempts to revert the user back to Sponsor, cannot because there are now 1250 Sponsors.

This happens to somewhere between 10-15 people. Due to people requesting refunds, we've been able to revert all but ~5 of them at this point.
